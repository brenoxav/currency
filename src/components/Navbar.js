import { NavLink } from 'react-router-dom';
import '../styles/Navbar.css';

const Navbar = () => (
  <header className="navbar">
    <NavLink exact to="/">
      <span>back</span>
    </NavLink>
    <span>forex today</span>
    <span>
      <span>mic</span>
      <span>opt</span>
    </span>
  </header>
);

export default Navbar;
