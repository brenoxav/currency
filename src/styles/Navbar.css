.navbar {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;
  background-color: var(--color-primary-dark);
}
