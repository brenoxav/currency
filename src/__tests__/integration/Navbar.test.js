import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Navbar from '../../components/Navbar';

describe('INTEGRATION TESTS IN NAVBAR', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Navbar />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the navbar heading', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    const heading = screen.getByText(/FOREX TODAY/i);

    expect(heading).toBeInTheDocument();
  });
});
