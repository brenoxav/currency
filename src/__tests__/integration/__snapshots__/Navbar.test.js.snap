// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INTEGRATION TESTS IN NAVBAR matches the snapshot 1`] = `
<header
  className="navbar"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="material-icons-outlined"
    >
      arrow_back_ios
    </span>
  </a>
  <span
    className="navbar-title"
  >
    forex today
  </span>
  <span>
    <span
      className="material-icons-outlined navbar-icons"
    >
      mic
    </span>
    <span
      className="material-icons-outlined navbar-icons"
    >
      settings
    </span>
  </span>
</header>
`;
